flocker/control/_model.py

```diff
@@ -689,6 +730,57 @@ def __new__(cls, **kwargs):
     def update_cluster_state(self, cluster_state):
         return cluster_state.update_node(self)
 
+    # Attributes that may be set to None to indicate ignorance:
+    _POTENTIALLY_IGNORANT_ATTRIBUTES = ["used_ports", "applications",
+                                        "manifestations", "paths",
+                                        "devices"]
+
+    def _completely_ignorant(self):
+        """
+        Return whether the node is completely ignorant of anything.
+        """
+        return all(getattr(self, attr) is None
+                   for attr in self._POTENTIALLY_IGNORANT_ATTRIBUTES)
+
+    def get_information_wipe(self):
+        attributes = [attr for attr in
+                      self._POTENTIALLY_IGNORANT_ATTRIBUTES
+                      if getattr(self, attr) is not None]
+        return _WipeNodeState(node_uuid=self.uuid, attributes=attributes)
+
+
+@implementer(IClusterStateWipe)
+class _WipeNodeState(PRecord):
+    """
+    Wipe information about a specific node from a ``DeploymentState``.
+
+    Only specific attributes will be wiped. If all attributes have been
+    wiped off the relevant ``NodeState`` then it will also be removed from
+    the ``DeploymentState`` completely.
+
+    :ivar UUID node_uuid: The UUID of the node being wiped.
+    :ivar PSet attributes: Names of ``NodeState`` attributes to wipe.
+    """
+    node_uuid = field(mandatory=True, type=UUID)
+    attributes = pset_field(str)
+
+    def update_cluster_state(self, cluster_state):
+        nodes = {n for n in cluster_state.nodes
+                 if n.uuid == self.node_uuid}
```

Comment:
Hm too bad `cluster_state.get_node` doesn't support this.  File a follow-up issue for making it accept a default to return instead of creating one?